### ISSUE TITLE
[BUG] Infinite retries when max retry is set to 0.

### ISSUE THREAD

**Issue by mutayTorre** [langchain4j/langchain4j#928]
**Describe the bug**
I tried setting the max retry to zero to prevent automatic retries when there is an error. It seems setting it to 0, doesn't stop it from retrying, instead it retries till infinity.

**Log and Stack trace**
```
d.l.i.RetryUtils - Exception was thrown on attempt 18 of 0
dev.ai4j.openai4j.OpenAiHttpException: {
  "error": {
    "message": "An assistant message with 'tool_calls' must be followed by tool messages responding to each 'tool_call_id'. The following tool_call_ids did not have response messages: call_ZuTo6iUVTy0Vuw6R1u09mPgK",
    "type": "invalid_request_error",
    "param": "messages.[5].role",
    "code": null
  }
}

        at dev.ai4j.openai4j.Utils.toException(Utils.java:8)
        at dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:28)
        at dev.ai4j.openai4j.RequestExecutor.execute(RequestExecutor.java:59)
        at dev.langchain4j.model.openai.OpenAiChatModel.lambda$generate$1(OpenAiChatModel.java:148)
        at dev.langchain4j.internal.RetryUtils$RetryPolicy.withRetry(RetryUtils.java:192)
        at dev.langchain4j.internal.RetryUtils.withRetry(RetryUtils.java:229)
        at dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:148)
        at dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:113)
        at dev.langchain4j.service.DefaultAiServices$1.invoke(DefaultAiServices.java:130)
        at agents.$Proxy44.chat(Unknown Source)
```

**To Reproduce**
Set the `maxRetries` to 0 when building an `OpenAiChatModel` and then try to send a message to the LLM that will trigger an error. 

**Expected behavior**
It shouldn't retry when the `maxRetries` is set to 0.

**Please complete the following information:**
- LangChain4j version: 0.28.0
- LLM(s) used: GPT_4_1106_PREVIEW
- Java version:  Java 11

**Additional context**

**Comment by langchain4j** [comment]
Thanks a lot for reporting!

**Comment by tess1o** [comment]
https://github.com/langchain4j/langchain4j/blob/03528af8a1de431a506b8f44947865b11c2b4607/langchain4j-core/src/main/java/dev/langchain4j/internal/RetryUtils.java#L188

```
        /**
         * This method attempts to execute a given action up to a specified number of times with a 1-second delay.
         * If the action fails on all attempts, it throws a RuntimeException.
         *
         * @param action      The action to be executed.
         * @param maxAttempts The maximum number of attempts to execute the action.
         * @param <T> The type of the result of the action.
         * @return The result of the action if it is successful.
         * @throws RuntimeException if the action fails on all attempts.
         */
        public <T> T withRetry(Callable<T> action, int maxAttempts) {
            int attempt = 1;
            while (true) {
                try {
                    return action.call();
                } catch (Exception e) {
                    if (attempt == maxAttempts) {
                        throw new RuntimeException(e);
                    }

                    log.warn(format("Exception was thrown on attempt %s of %s", attempt, maxAttempts), e);

                    sleep(attempt);
                }
                attempt++;
            }
        }
```
This method assumes that maxAttempts is always > 0 and thus the loop never ends. As a workaround the user can set maxRetries to 1.

By the way, if maxRetries  is set to 1, it means that the original callable will be executed only once (so it won't be retried). I don't think it's a correct behavior. With maxRetries = 1 the original callable should be executed two times (one time as "original call", the second time as a retry).

**Comment by VaderKai** [comment]
Hi, I'm interested in the project and would like to start contributing to it,can you assign this issue to me?

**Comment by langchain4j** [comment]
Fixed by #981
